src/OmniSharp.Roslyn.CSharp/Services/QuickInfoProvider.cs

```diff
@@ -0,0 +1,197 @@
+        /// list.
+        /// </summary>
+        private const string ContainerStart = nameof(ContainerStart);
+        /// <summary>
+        /// Indicates the end of a text container. See <see cref="ContainerStart"/>.
+        /// </summary>
+        private const string ContainerEnd = nameof(ContainerEnd);
+
+        private readonly OmniSharpWorkspace _workspace;
+        private readonly FormattingOptions _formattingOptions;
+
+        [ImportingConstructor]
+        public QuickInfoProvider(OmniSharpWorkspace workspace, FormattingOptions formattingOptions)
+        {
+            _workspace = workspace;
+            _formattingOptions = formattingOptions;
+        }
+
+        public async Task<QuickInfoResponse> Handle(QuickInfoRequest request)
+        {
+            var document = _workspace.GetDocument(request.FileName);
+            var response = new QuickInfoResponse();
+
+            if (document is null)
+            {
+                return response;
+            }
+
+            var quickInfoService = QuickInfoService.GetService(document);
+            if (quickInfoService is null)
+            {
+                return response;
+            }
+
+            var sourceText = await document.GetTextAsync();
+            var position = sourceText.Lines.GetPosition(new LinePosition(request.Line, request.Column));
+
+            var quickInfo = await quickInfoService.GetQuickInfoAsync(document, position);
+            if (quickInfo is null)
+            {
+                return response;
+            }
+
+
+            var sb = new StringBuilder();
+            response.Description = quickInfo.Sections.FirstOrDefault(s => s.Kind == QuickInfoSectionKinds.Description)?.Text;
+
+            var documentation = quickInfo.Sections.FirstOrDefault(s => s.Kind == QuickInfoSectionKinds.DocumentationComments);
+            if (documentation is object)
+            {
+                response.Summary = getMarkdown(documentation.TaggedParts);
+            }
+
+            response.RemainingSections = quickInfo.Sections
+                                                  .Where(s => s.Kind != QuickInfoSectionKinds.Description && s.Kind != QuickInfoSectionKinds.DocumentationComments)
+                                                  .Select(s =>
+                                                  {
+                                                      switch (s.Kind)
+                                                      {
+                                                          case QuickInfoSectionKinds.AnonymousTypes:
+                                                          case QuickInfoSectionKinds.TypeParameters:
+                                                              return new QuickInfoResponseSection { IsCSharpCode = true, Text = s.Text };
+
+                                                          default:
```

Comment:
Should we special case any `example` as well and emit them as full markdown code blocks?

Reply:
I don't believe there is an examples section in quickinfo.